proposals/pubsub.md

```diff
@@ -0,0 +1,182 @@
+<!--
+Outline the status quo, including any relevant context on the problem you’re seeing that this project should solve.
+Wherever possible, include pains or problems that you’ve seen users experience to help motivate why solving this problem works towards top-line objectives.
+-->
+
+As of this writing [pubsub][IPFS pubsub] is considered experimental and is disabled in go-ipfs.
+
+> Can be enabled by starting a daemon with `--enable-pubsub-experiment` flag
+
+In practice however many teams building products on IPFS use and depend on pubsub:
+
+- Textile nodes have pubsub enabled.
+- Ceramic uses pubsub for queries and log replication.
+- [OrbitDB][] depends on pubsub to do replication.
+- Fission team would like to use pubsub (has user requests, but found it unreliable)
+
+It also fragments network by (JS) nodes having it enabled and nodes that do not making IPNS unrelaible.
+
+> IPFS in browsers use IPNS over pubsub (no DHT for browsers) while other nodes do it over DHT.
+
+Enabling PubSub would populate network with more nodes making it more reliable.
+
+
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+<!--(bullet list)-->
+
+- Teams building products need pubsub.
+- Disabled pubsub makes an ecosystem more complex.
+- Building (near) real-time application on IPFS requires pubsub.
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+<!--(short paragraph)-->
+
+They would just be able to use `pubsub` without `--enable-pubsub-experiment` flag. Feature becomes usable to ipfs-client's that use IPFS HTTP API.
+
+> **From Fission**:
+> 
+> If we expect apps to be independent and run on the IPFS network -- like they do with HTTP servers -- then whatever else can be made available natively can be powerful building blocks
+>
+> Having more capabilities available out of the box for developers is a better reason to build and host on IPFS natively. Pubsub out of the box gives a host of real time and messaging opportunities that our customers would love to use.
+
+#### Impact
+_How would this directly contribute to web3 dev stack product-market fit?_
+
+<!--
+Explain how this addresses known challenges or opportunities.
+What awesome potential impact/outcomes/results will we see if we nail this project?
+-->
+
+- PubSub becomes more reliable with more nodes on the network.
+- Bulding (near) real-time products becomes easier, IPFS nodes are suitable out of the box.
+
+#### Leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+<!--
+Explain the opportunity or leverage point for our subsequent velocity/impact (e.g. by speeding up development, enabling more contributors, etc)
+-->
+
+- Improves velocity of teams building products requiring pubsub by removing additional burden.
+- Improves pubsub reliability by adding more enabled nodes to the network.
```

Comment:
The number of pubsub enabled nodes on the network doesn't make pubsub more reliable as nodes only participate in topics they're subscribed to.

Reply:
Note: more nodes supporting and using ipns-over-pubsub would help, but only nodes resolving a specific IPNS address would participate in that IPNS address' swarm.